aws_lambda_powertools/utilities/feature_flags/schema.py

```diff
@@ -264,9 +290,79 @@ def validate_condition_key(condition: Dict[str, Any], rule_name: str):
+            )
+        if action == RuleAction.SCHEDULE_BETWEEN_DATETIME_RANGE.value and key != TimeKeys.CURRENT_DATETIME_UTC.value:
+            raise SchemaValidationError(
+                f"'condition with a 'SCHEDULE_BETWEEN_DATETIME_RANGE' action must have a 'CURRENT_DATETIME_UTC' condition key, rule={rule_name}"  # noqa: E501
+            )
+        if action == RuleAction.SCHEDULE_BETWEEN_DAYS_OF_WEEK.value and key != TimeKeys.CURRENT_DAY_UTC.value:
+            raise SchemaValidationError(
+                f"'condition with a 'SCHEDULE_BETWEEN_DAYS_OF_WEEK' action must have a 'CURRENT_DAY_UTC' condition key, rule={rule_name}"  # noqa: E501
+            )
 
     @staticmethod
     def validate_condition_value(condition: Dict[str, Any], rule_name: str):
         value = condition.get(CONDITION_VALUE, "")
         if not value:
             raise SchemaValidationError(f"'value' key must not be empty, rule={rule_name}")
+        action = condition.get(CONDITION_ACTION, "")
+        # time actions
+        if action == RuleAction.SCHEDULE_BETWEEN_TIME_RANGE.value:
+            ConditionsValidator._validate_schedule_between_time_and_datetime_ranges(
+                value, rule_name, action, ConditionsValidator.TIME_RANGE_FORMAT
+            )
+        elif action == RuleAction.SCHEDULE_BETWEEN_DATETIME_RANGE.value:
+            ConditionsValidator._validate_schedule_between_time_and_datetime_ranges(
+                value, rule_name, action, ConditionsValidator.DATETIME_RANGE_FORMAT
+            )
+        elif action == RuleAction.SCHEDULE_BETWEEN_DAYS_OF_WEEK.value:
+            ConditionsValidator._validate_schedule_between_days(value, rule_name)
+
+    @staticmethod
+    def _validate_time_value(time: str, rule_name: str, date_format: str):
+        try:
+            datetime.strptime(time, date_format)
+        except Exception:
+            raise SchemaValidationError(
+                f"'START' and 'END' must be a valid time format, time_format={date_format}, rule={rule_name}"
+            )
+
+    @staticmethod
+    def _validate_schedule_between_days(value: Any, rule_name: str):
+        if not isinstance(value, list) or not value:
+            raise SchemaValidationError(
+                f"condition with a CURRENT_DAY_UTC action must have a non empty condition value type list, rule={rule_name}"  # noqa: E501
+            )
+        for day in value:
+            if not isinstance(day, str) or day not in [
+                TimeValues.SUNDAY.value,
+                TimeValues.MONDAY.value,
+                TimeValues.TUESDAY.value,
+                TimeValues.WEDNESDAY.value,
+                TimeValues.THURSDAY.value,
+                TimeValues.FRIDAY.value,
+                TimeValues.SATURDAY.value,
+            ]:
+                raise SchemaValidationError(
+                    f"condition value must represent a week day string defined in 'TimeValues' enum, rule={rule_name}"
```

Comment:
week day for me is Monday - Frday

```suggestion
                    f"condition value must represent a day of the week string defined in 'TimeValues' enum, rule={rule_name}"
```

Reply:
i'll align it to match most of the world. we are special, it's my day off here.
starting with monday.

Reply:
The point here was about "week day" vs "day of the week", not which days are work days :P

Reply:
lol

Reply:
fixed